store/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,719 @@
+package dynamodb
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/abronan/valkeyrie"
+	"github.com/abronan/valkeyrie/store"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+)
+
+const (
+	// DefaultReadCapacityUnits default read capacity used to create table
+	DefaultReadCapacityUnits = 2
+	// DefaultWriteCapacityUnits default write capacity used to create table
+	DefaultWriteCapacityUnits = 2
+	// TableCreateTimeoutSeconds the maximum time we wait for the AWS DynamoDB table to be created
+	TableCreateTimeoutSeconds = 30
+	// DeleteTreeTimeoutSeconds the maximum time we retry a write batch
+	DeleteTreeTimeoutSeconds = 30
+
+	partitionKey          = "id"
+	revisionAttribute     = "version"
+	encodedValueAttribute = "encoded_value"
+	ttlAttribute          = "expiration_time"
+	lockAttribute         = "lock"
+
+	noExpiration   = time.Duration(0)
+	defaultLockTTL = 60 * time.Second
+)
+
+var (
+	// ErrBucketOptionMissing is returned when bucket config option is missing
+	ErrBucketOptionMissing = errors.New("missing dynamodb bucket/table name")
+	// ErrMultipleEndpointsUnsupported is returned when more than one endpoint is provided
+	ErrMultipleEndpointsUnsupported = errors.New("dynamodb only supports one endpoint")
+	// ErrTableCreateTimeout table creation timed out
+	ErrTableCreateTimeout = errors.New("dynamodb table creation timed out")
+	// ErrDeleteTreeTimeout delete batch timed out
+	ErrDeleteTreeTimeout = errors.New("delete batch timed out")
+	// ErrLockAcquireCancelled stop called before lock was acquired.
+	ErrLockAcquireCancelled = errors.New("stop called before lock was acquired")
+)
+
+// Register register a store provider in valkeyrie for AWS DynamoDB
+func Register() {
+	valkeyrie.AddStore(store.DYNAMODB, New)
+}
+
+// New opens a creates a new table
```

Comment:
```suggestion
// New opens and creates a new table
```

Reply:
@wolfeidau After testing I noticed that `createTable` is not used and the table should be created by hand. It is not that I don't like it, but what did you miss a call or it is intended and the comment is not correct?

Reply:
Thanks for fixing the typo @wolfeidau, but what about my question, should the table be created in the call to `New`?